users/vieting/experiments/switchboard/ctc/feat/network_helpers/specaug_configurable.py

```diff
@@ -0,0 +1,300 @@
+from typing import Dict, Union, Optional, List
+
+
+def add_specaug_layer(
+    network: Dict,
+    name: str = "specaug",
+    from_list: Optional[Union[str, List[str]]] = None,
+    num_epochs: int = 450,
+    config: Dict = None,
+) -> tuple:
+    """
+    Add a customizable SpecAugment layer to the network.
+
+    This function adds a highly configurable SpecAugment layer to the given network,
+    allowing for fine-grained control over various aspects of the augmentation process.
+
+    Args:
+        network (Dict): The network to which the SpecAugment layer will be added.
+        name (str, optional): The name of the SpecAugment layer. Defaults to "specaug".
+        from_list (Optional[Union[str, List[str]]], optional): The input layer(s) for the SpecAugment layer.
+            Default is ["data"].
+        config (Dict, optional): A dictionary containing configuration options for the SpecAugment layer.
+            Config Dict Parameters:
+                max_time_num (int): The beginning maximum number of time masks to be applied. Default is 3.
+                max_time (int): The beginning maximum size (in time frames) of each time mask. Default is 10.
+                max_feature_num (int): The beginning maximum number of frequency masks to be applied. Default is 4.
+                max_feature (int): The beginning maximum size (in frequency bins) of each frequency mask. Default is 5.
+                enable_sorting (bool): Whether to sort filters by their center frequency before applying masks. Default is True.
+                steps_per_epoch (int): The number of steps per epoch. Make sure this parameter is accurate since the all the scheduling depends on it. Default is 2080.
```

Comment:
This is dangerous, it can easily be overlooked and that would introduce unexpected behaviour. I suggest to force configuring this.

Reply:
I do not know if an how it would be possible to force an entry in a dict.
I could add an assertion or remove the parameter from the dict and add it as a separate parameter.
An assertion seems better because it would be unintuitive if just one of the specaugment parameter needs to be set separately from the others.

Reply:
Yes, just remove the default and assert that the key exists.